### PR TITLE
Add a note about docker installed from snap

### PR DESCRIPTION
**Pull request by dsame**
### Why:

One of user had a problem with the docker in self-hosted environment. The investigation showed it used the engine installed with snap and that was a reason of the problem.

Original issue: https://github.com/actions/setup-java/issues/348

### What's being changed (if available, include any code snippets, screenshots, or gifs):

I added a sectionDocker engine setup

### Check off the following:

- [ x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Review by marko-zivic-93 — DISMISSED** [review]

**Comment by cmwilson21** [comment]
@dsame Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/actions/using-github-hosted-runners/about-github-hosted-runners.md [hunk 1]
```diff
@@ -190,6 +190,10 @@ Actions that run in Docker containers have static directories under the `/github
 - `/github/workspace` - {% data reusables.repositories.action-root-user-required %}
 - `/github/workflow`
 
+### Docker engine setup
+
+To pass the inputs of an action into the docker container the runner uses the environment variables which may contain dashes as part of their names. If  the docker engine is not a binary executable but a shell wrapper or a link (a docker engine istalled on Linux with snap is a sample) the action is not able to get the inputs and the build fails with the message `Error: Input required and not supplied: java-version`.
```
Comment (martin389) [inline content/actions/using-github-hosted-runners/about-github-hosted-runners.md:195]:
```suggestion
To pass the inputs of an action into the docker container, the runner uses the environment variables, which may contain dashes as part of their names. The action may not able to get the inputs if the Docker engine is not a binary executable, and is instead a shell wrapper or a link (for example, a Docker engine installed on Linux using `snap). If this occurs, the build will subsequently fail with the message `Error: Input required and not supplied: java-version`.
```

**Review by martin389 — COMMENTED** [review]
Thanks for the PR 👍 I've add some small edits in a suggestion. You mentioned this occurred in a self-hosted runner environment, so would it make sense to move this section to the self-hosted runner docs "[Customizing the containers used by jobs](https://docs.github.com/en/actions/hosting-your-own-runners/customizing-the-containers-used-by-jobs)"? This is the applicable file: https://github.com/github/docs/blob/main/content/actions/hosting-your-own-runners/customizing-the-containers-used-by-jobs.md

File: content/actions/using-github-hosted-runners/about-github-hosted-runners.md (hunk 1 above)
Comment (martin389) [inline content/actions/using-github-hosted-runners/about-github-hosted-runners.md:195]:
```suggestion
To pass the inputs of an action into the Docker container, the runner uses environment variables that might contain dashes as part of their names. If the Docker engine is not a binary executable but a shell wrapper or a link (for example, a Docker engine installed on Linux with `snap`) the action is not able to get the inputs, and the build consequently fails with the message `Error: Input required and not supplied: java-version`.
```

File: content/actions/using-github-hosted-runners/about-github-hosted-runners.md [hunk 2]
```diff
@@ -199,6 +199,10 @@ Actions that run in Docker containers have static directories under the `/github
 - `/github/workspace` - {% data reusables.repositories.action-root-user-required %}
 - `/github/workflow`
 
+### Docker engine setup
+
+To pass the inputs of an action into the Docker container, the runner uses environment variables that might contain dashes as part of their names. If the Docker engine is not a binary executable but a shell wrapper or a link (for example, a Docker engine installed on Linux with `snap`) the action is not able to get the inputs, and the build consequently fails with the message `Error: Input required and not supplied: java-version`.
```
Comment (martin389) [inline content/actions/using-github-hosted-runners/about-github-hosted-runners.md:204]:
```suggestion
To pass the inputs of an action into the Docker container, the runner uses environment variables that might contain dashes as part of their names. The action may not able to get the inputs if the Docker engine is not a binary executable, but is instead a shell wrapper or a link (for example, a Docker engine installed on Linux using `snap). If this occurs, the build will subsequently fail with the message `Error: Input required and not supplied: java-version`.
```

File: content/actions/using-github-hosted-runners/about-github-hosted-runners.md [hunk 3]
```diff
@@ -199,6 +199,10 @@ Actions that run in Docker containers have static directories under the `/github
 - `/github/workspace` - {% data reusables.repositories.action-root-user-required %}
 - `/github/workflow`
 
+### Docker engine setup
+
+To pass the inputs of an action into the Docker container, the runner uses environment variables that might contain dashes as part of their names. The action may not able to get the inputs if the Docker engine is not a binary executable, but is instead a shell wrapper or a link (for example, a Docker engine installed on Linux using `snap). If this occurs, the build will subsequently fail with the message `Error: Input required and not supplied: java-version`.
```
Comment (martin389) [inline content/actions/using-github-hosted-runners/about-github-hosted-runners.md:204]:
```suggestion
### Checking how Docker was installed

To pass the inputs of an action into the Docker container, the runner uses environment variables that might contain dashes as part of their names. The action may not able to get the inputs if the Docker engine is not a binary executable, but is instead a shell wrapper or a link (for example, a Docker engine installed on Linux using `snap`). If this occurs, the build will subsequently fail with the message `Error: Input required and not supplied: java-version`.
```

**Review by panticmilos — APPROVED** [review]

**Comment by martin389** [comment]
Addressing this in an internal PR, so that it can be moved to the self-hosted runner troubleshooting section.

**Comment by martin389** [comment]
These changes have now shipped in a separate PR. The content will be live here in a few minutes: https://docs.github.com/en/actions/hosting-your-own-runners/monitoring-and-troubleshooting-self-hosted-runners#checking-which-docker-engine-is-installed-on-the-runner

Thanks for the PR, @dsame 👍

**Pull request closed without merging**